lms/djangoapps/lms_xblock/mixin.py

```diff
@@ -142,6 +144,62 @@ def _get_user_partition(self, user_partition_id):
 
         raise NoSuchUserPartitionError("could not find a UserPartition with ID [{}]".format(user_partition_id))
 
+    def get_parent_unit(self):
+        """
+        Finds xblock's parent unit if it exists.
+
+        Returns:
+            xblock: Returns the parent unit xblock if it exists.
+            If no parent unit exists, returns None.
+        """
+        xblock = self
+        while xblock:
+            xblock = xblock.get_parent()
```

Comment:
And it looks like there is an existing helper method you can use, in lms/djangoapps/edxnotes/helpers.py, get_parent_unit. It's more like your first version, but I think it is preferable to have only 1 version.

I'm not sure where to put it though.... Maybe a new utils file-- lms/lib/utils.py?